spikeinterface/extractors/stream_ibl.py

```diff
@@ -0,0 +1,204 @@
+        probe_labels = set([raw_content.split("/")[1] for raw_content in raw_contents])
+
+        stream_names = list()
+        for probe_label in probe_labels:
+            raw_suffixes_by_probe = set(
+                [Path(raw_content).suffixes[-2] for raw_content in raw_contents if "probe00" in raw_content]
+            )
+            if ".ap" in raw_suffixes_by_probe:
+                stream_names.append(probe_label + ".ap")
+            if ".lf" in raw_suffixes_by_probe:
+                stream_names.append(probe_label + ".lf")
+
+        return stream_names
+
+    def __init__(
+        self,
+        session: str,
+        stream_name: str,
+        load_sync_channel: bool = False,
+        cache_folder: Optional[Union[Path, str]] = None,
+        remove_cached: bool = True,
+    ):
+        from brainbox.io.spikeglx import Streamer
+        from one.api import ONE
+        from neo.rawio.spikeglxrawio import read_meta_file, extract_stream_info
+
+        one = ONE(base_url="https://openalyx.internationalbrainlab.org", password="international", silent=True)
+
+        session_names = self.get_stream_names(session=session)
+        assert stream_name in session_names, (
+            f"The `stream_name` '{stream_name}' was not found in the available listing for session '{session}'! "
+            f"Please choose one of {session_names}."
+        )
+        probe_label, stream_type = stream_name.split(".")
+
+        insertions = one.alyx.rest("insertions", "list", session=session)
+        pid = next(insertion["id"] for insertion in insertions if insertion["name"] == probe_label)
+
+        cache_folder = Path(cache_folder) if cache_folder is not None else cache_folder
+        file_streamer = Streamer(
+            pid=pid, one=one, typ=stream_type, cache_folder=cache_folder, remove_cached=remove_cached
+        )
+
+        # get basic metadata
+        meta_file = file_streamer.file_meta_data  # streamer downloads uncompressed metadata files on init
+        meta = read_meta_file(meta_file)
+        info = extract_stream_info(meta_file, meta)
+        channel_ids = info["channel_names"]
+        gains = info["channel_gains"]
+        offsets = info["channel_offsets"]
+        if not load_sync_channel:
+            channel_ids = channel_ids[:-1]
+            gains = gains[:-1]
+            offsets = offsets[:-1]
+
+        # initialize main extractor
+        sampling_frequency = file_streamer.fs
+        dtype = file_streamer.dtype
+        BaseRecording.__init__(self, channel_ids=channel_ids, sampling_frequency=sampling_frequency, dtype=dtype)
+        self.extra_requirements.append("ONE-api")
+        self.extra_requirements.append("ibllib")
+        self.set_channel_gains(gains)
+        self.set_channel_offsets(offsets)
+
+        # set probe
+        probe = pi.read_spikeglx(meta_file)
+        self.set_probe(probe, in_place=True)
```

Comment:
I load_sync_channel is True this ìnplace` is not possible because contact size will not the channel size. (this is a requirement for the `inplace`)
We need to protect this with 
```suggestion
        if not load_sync_channels:
            probe = pi.read_spikeglx(meta_file)
            self.set_probe(probe, in_place=True)
```

Reply:
Sure, would you need it [here](https://github.com/catalystneuro/spikeinterface/blob/streaming_ibl_extractor/spikeinterface/extractors/cbin_ibl.py#L86-L87) in the CBin code too then? The meta files ought to be exactly the same

Reply:
I notice the core SpikeGLX framework also does something different as well: https://github.com/catalystneuro/spikeinterface/blob/streaming_ibl_extractor/spikeinterface/extractors/neoextractors/spikeglx.py#L69-L73

Reply:
OK added it here and to the CBIN as well